src/components/calendar/dateUtil.js

```diff
@@ -184,7 +185,26 @@
      * @return {boolean} Whether the date is a valid Date.
      */
     function isValidDate(date) {
-      return date != null && date.getTime && !isNaN(date.getTime());
+      return date !== null && date.getTime && !isNaN(date.getTime());
+    }
+
+    /**
+     * Creates a date with the time set to midnight.
+     * Drop-in replacement for two forms of the Date constructor:
+     * 1. No argument for Date representing now.
+     * 2. Single-argument value representing number of seconds since Unix Epoch.
+     * @param {number=} opt_value
+     * @return {Date} New date with time set to midnight.
+     */
+    function createDateAtMidnight(opt_value) {
+      var date;
+      if (typeof opt_value !== 'undefined') {
```

Comment:
You can use `angular.isUndefined`
https://docs.angularjs.org/api/ng/function/angular.isUndefined